dotnet/src/Agents/Core/Chat/KernelFunctionTerminationStrategy.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// Signals termination based on the evaluation of a <see cref="KernelFunction"/>.
+/// </summary>
+/// <param name="function">A <see cref="KernelFunction"/> used for termination criteria</param>
+public class KernelFunctionTerminationStrategy(KernelFunction function) : TerminationStrategy
+{
+    /// <summary>
+    /// A well-known <see cref="KernelArguments"/> key associated with the agent name.
+    /// </summary>
+    public const string ArgumentKeyAgent = "_agent_";
```

Comment:
AgentArgumentName?

Reply:
FWIW - There is a school of thought where logical naming proceeds in the order of general to specific.  (Ball Red vs Red Ball).